pyproj/_crs.pyx

```diff
@@ -295,6 +295,10 @@ cdef class Base:
         if self.context != NULL:
             proj_context_destroy(self.context)
 
+    def _ensure_initialized(self):
+        if self.projobj == NULL:
+            raise CRSError("Object not initialized.")
```

Comment:
Alternatively to doing those checks in all methods, can't we simply disallow creating an uninitialized object?

So that it already raises an error if you do `d = Datum()` ?

Reply:
We need to have that ability in the `create` methods:
https://github.com/pyproj4/pyproj/blob/aab1acbdf4fa88149a6c3c30dde950e3af01ef8b/pyproj/_crs.pyx#L451

So, this is the next best thing I could think of.

Reply:
Looking for initialization in cython, I found in their docs:

> ... use a factory function instead that handles the object initialisation. It often helps to directly call `__new__()` in this function to bypass the call to the `__init__()` constructor.

https://cython.readthedocs.io/en/latest/src/userguide/special_methods.html

So that would also be an option I think

Reply:
> Looking for initialization in cython, I found in their docs:
> 
> > ... use a factory function instead that handles the object initialisation. It often helps to directly call `__new__()` in this function to bypass the call to the `__init__()` constructor.
> 
> https://cython.readthedocs.io/en/latest/src/userguide/special_methods.html
> 
> So that would also be an option I think

That sounds like a much more secure/simple method for this. I will give it a go later.